evcache-core/src/main/java/net/spy/memcached/EVCacheMemcachedClient.java

```diff
@@ -111,6 +113,11 @@ private boolean isWrongKeyReturned(String original_key, String returned_key) {
             log.error("Wrong key returned. Key - " + original_key + " (Host: " + original_host + ") ; Returned Key "
                         + returned_key + " (Host: " + returned_host + ")", new Exception());
             client.reportWrongKeyReturned(original_host);
+
+            // If we are configured to dynamically switch log levels to DEBUG on a wrong key error, do so here.
+            if (autoLogSwitchOnWrongKey.get()) {
+                org.apache.log4j.Logger.getRootLogger().setLevel(org.apache.log4j.Level.toLevel("DEBUG"));
```

Comment:
Which logger is the root logger in this context? We want to limit debug logs to "log4j.logger.net.spy.memcached" or know the name of the logger we are enabling here in order to disable it.

Reply:
@tharanga This is a little unclear to me as well. Would something like this work?
org.apache.log4j.Logger.getLogger(EVCacheMemcachedClient.class).setLevel(org.apache.log4j.Level.toLevel("DEBUG"));

But this might only make logs DEBUG for the EVCacheMemcachedClient class.

Reply:
I changed it to ```System.setProperty("log4j.logger.net.spy", "DEBUG");``` as per @arunagrawal84's suggestion.